### ISSUE TITLE
Unexpected KeyErrors when scrapping and logged in.

### ISSUE THREAD

**Issue by bschollnick** [instaloader/instaloader#433]
**Describe the bug**
This appears to be related to  #380, if I scrap anything, when logged in roughly 75% of the time the session does not finish, and ends with KeyError...  (I'm working to compile the list but it's GraphQ?  and at least 3-4 other errors.

**To Reproduce**
```
instaloader <Instagram username> \
    --sessionfile=/users/benjamin/IL_sessionfile.txt \
    --no-videos --no-compress-json  --no-metadata-json --login USERNAME -p PASSWORD
```

**Expected behavior**
Success fetch without errors.

**Error messages and tracebacks**
```
Loaded session from /users/benjamin/IL_sessionfile.txt.
Logged in as godzillahentai234.
fay198978/2019-03-26_19-02-51_UTC_profile_pic.jpg already exists
Retrieving posts from profile fay198978.
[  1/192] fay198978/2019-10-25_02-24-58_UTC.jpg [深夜，涉谷，GANTZ。  #gantz #gantzco…] 
[  2/192] fay198978/2019-10-24_11-38-43_UTC.jpg exists [走走走，打怪去。 #gantz #gantzcosplay] unchanged 
[  3/192] fay198978/2019-10-24_07-24-16_UTC.jpg exists [「爆汗吧濺身少女」寫真本打樣出來了！🎉 11/2 圓山花博…] unchanged Traceback (most recent call last):
  File "/Users/benjamin/venvs/downloader/lib/python3.8/site-packages/instaloader/structures.py", line 157, in _field
    d = d[key]
KeyError: 'video_url'

Traceback (most recent call last):
  File "/Users/benjamin/venvs/downloader/bin/instaloader", line 8, in <module>
    sys.exit(main())
  File "/Users/benjamin/venvs/downloader/lib/python3.8/site-packages/instaloader/__main__.py", line 401, in main
    _main(loader,
  File "/Users/benjamin/venvs/downloader/lib/python3.8/site-packages/instaloader/__main__.py", line 187, in _main
    instaloader.download_profiles(profiles,
  File "/Users/benjamin/venvs/downloader/lib/python3.8/site-packages/instaloader/instaloader.py", line 1039, in download_profiles
    downloaded = self.download_post(post, target=profile_name)
  File "/Users/benjamin/venvs/downloader/lib/python3.8/site-packages/instaloader/instaloader.py", line 491, in download_post
    downloaded &= self.download_pic(filename=filename, url=post.video_url, mtime=post.date_local)
  File "/Users/benjamin/venvs/downloader/lib/python3.8/site-packages/instaloader/structures.py", line 290, in video_url
    return self._field('video_url')
  File "/Users/benjamin/venvs/downloader/lib/python3.8/site-packages/instaloader/structures.py", line 160, in _field
    d = self._full_metadata
  File "/Users/benjamin/venvs/downloader/lib/python3.8/site-packages/instaloader/structures.py", line 143, in _full_metadata
    self._obtain_metadata()
  File "/Users/benjamin/venvs/downloader/lib/python3.8/site-packages/instaloader/structures.py", line 135, in _obtain_metadata
    self._full_metadata_dict = pic_json['entry_data']['PostPage'][0]['graphql']['shortcode_media']
KeyError: 'graphql'


Retrieving posts from profile arabella_kat.
[  1/ 76] Traceback (most recent call last):
  File "/Users/benjamin/venvs/downloader/lib/python3.8/site-packages/instaloader/structures.py", line 157, in _field
    d = d[key]
KeyError: 'edge_sidecar_to_children'
```

These appear to be the main key errors...

**Instaloader version**
```
(downloader) benjamin@Nerv dropbox % instaloader --version
4.2.9
```

**Additional context**
Python v3.8 Stable.

**Comment by mdPlusPlus** [comment]
Running 4.2.8 resulted in `KeyError: 'edge_media_to_parent_comment'` and `KeyError: 'edges'` for me. 4.2.9 added `KeyError: 'graphql'`.

**Comment by i-cant-git** [comment]
I also get this error when I try to retrieve posts.

```
$ instaloader --login username example
Loaded session from username.
Logged in as dogoncelik.
example/2019-10-25_16-04-33_UTC_profile_pic.jpg already exists
Retrieving posts from profile example.
[  1/ 60] Traceback (most recent call last):
  File "python3\python-3.6.1.amd64\lib\site-packages\instaloader\structures.py", line 157, in _field
    d = d[key]
KeyError: 'edge_sidecar_to_children'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "python3\python-3.6.1.amd64\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "python3\python-3.6.1.amd64\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "python3\python-3.6.1.amd64\Scripts\instaloader.exe\__main__.py", line 9, in <module>
  File "python3\python-3.6.1.amd64\lib\site-packages\instaloader\__main__.py", line 414, in main
    storyitem_filter_str=args.storyitem_filter)
  File "python3\python-3.6.1.amd64\lib\site-packages\instaloader\__main__.py", line 189, in _main
    download_stories, fast_update, post_filter, storyitem_filter)
  File "python3\python-3.6.1.amd64\lib\site-packages\instaloader\instaloader.py", line 1039, in download_profiles
    downloaded = self.download_post(post, target=profile_name)
  File "python3\python-3.6.1.amd64\lib\site-packages\instaloader\instaloader.py", line 466, in download_post
    for sidecar_node in post.get_sidecar_nodes():
  File "python3\python-3.6.1.amd64\lib\site-packages\instaloader\structures.py", line 227, in get_sidecar_nodes
    for edge in self._field('edge_sidecar_to_children', 'edges'):
  File "python3\python-3.6.1.amd64\lib\site-packages\instaloader\structures.py", line 160, in _field
    d = self._full_metadata
  File "python3\python-3.6.1.amd64\lib\site-packages\instaloader\structures.py", line 143, in _full_metadata
    self._obtain_metadata()
  File "python3\python-3.6.1.amd64\lib\site-packages\instaloader\structures.py", line 135, in _obtain_metadata
    self._full_metadata_dict = pic_json['entry_data']['PostPage'][0]['graphql']['shortcode_media']
KeyError: 'graphql'
```

**Comment by claudiofrancesconi** [comment]
sorry but i can't see any fix

**Comment by nagualcode** [comment]
I am having this issues also!

Retrieving saved posts...
[  1] Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/instaloader/structures.py", line 157, in _field
    d = d[key]
KeyError: 'edge_sidecar_to_children'

During handling of the above exception, another exception occurred:

**Comment by mdPlusPlus** [comment]
> https://github.com/instaloader/instaloader/blob/master/instaloader/instaloadercontext.py

> It seems it is probably on the folder you just run, inside of the "instaloader" folder should have a instaloadercontext.py file. I replaced mine with the one above and it worked just fine.

**Comment by elgallo** [comment]
still having an issue after updating the following

> > https://github.com/instaloader/instaloader/blob/master/instaloader/instaloadercontext.py
> 
> > It seems it is probably on the folder you just run, inside of the "instaloader" folder should have a instaloadercontext.py file. I replaced mine with the one above and it worked just fine.


this is my error:

```
[ 65/ 83] Traceback (most recent call last):
  File "/home/gallo/.local/lib/python3.6/site-packages/instaloader/structures.py", line 157, in _field
    d = d[key]
KeyError: 'edge_sidecar_to_children'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/gallo/.local/bin/instaloader", line 11, in <module>
    sys.exit(main())
  File "/home/gallo/.local/lib/python3.6/site-packages/instaloader/__main__.py", line 414, in main
    storyitem_filter_str=args.storyitem_filter)
  File "/home/gallo/.local/lib/python3.6/site-packages/instaloader/__main__.py", line 189, in _main
    download_stories, fast_update, post_filter, storyitem_filter)
  File "/home/gallo/.local/lib/python3.6/site-packages/instaloader/instaloader.py", line 1039, in download_profiles
    downloaded = self.download_post(post, target=profile_name)
  File "/home/gallo/.local/lib/python3.6/site-packages/instaloader/instaloader.py", line 466, in download_post
    for sidecar_node in post.get_sidecar_nodes():
  File "/home/gallo/.local/lib/python3.6/site-packages/instaloader/structures.py", line 227, in get_sidecar_nodes
    for edge in self._field('edge_sidecar_to_children', 'edges'):
  File "/home/gallo/.local/lib/python3.6/site-packages/instaloader/structures.py", line 160, in _field
    d = self._full_metadata
  File "/home/gallo/.local/lib/python3.6/site-packages/instaloader/structures.py", line 143, in _full_metadata
    self._obtain_metadata()
  File "/home/gallo/.local/lib/python3.6/site-packages/instaloader/structures.py", line 137, in _obtain_metadata
    if self.shortcode != self._full_metadata_dict['shortcode']:
TypeError: 'NoneType' object is not subscriptable
```

**to reproduce**

-l=userName -p=\*\*\* -s -F --no-compress-json --max-connection-attempts=2 -no-video-thumbnails --dirname-pattern=/path/to/ig --filename-pattern={target}_{date} [profile..profile22]

**Comment by aandergr** [comment]
Does this still occur with the recent 4.2.10 release?

**Comment by claudiofrancesconi** [comment]
> Does this still occur with the recent 4.2.10 release?

yes

**Comment by bschollnick** [comment]
I have not been able to duplicate it in 4.2.10.    I'll defer to your decision on either closing or keeping this open, since it sounds like a few other people have run into this?

**Comment by aandergr** [comment]
> I have not been able to duplicate it in 4.2.10. I'll defer to your decision on either closing or keeping this open, since it sounds like a few other people have run into this?

Thanks for the update! I close this issue for now since it looks exactly like #380 and #394 which have been fixed with #411 in the 4.2.10 release.

> > Does this still occur with the recent 4.2.10 release?
> 
> yes

@claudiofrancesconi Please open a new bug report with your exact error message, if you still run into problems, since it might have a different cause.
